JustSaying.Sample.Restaurant.KitchenConsole/Program.cs

```diff
@@ -1,55 +1,90 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
 using Amazon;
 using JustSaying.Sample.Restaurant.Models;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
-using System;
-using System.Threading.Tasks;
+[assembly: SuppressMessage("Reliability", "CA2007:Do not directly await a Task", Justification = "Console application do not have a SynchronizationContext")]
 
 namespace JustSaying.Sample.Restaurant.KitchenConsole
 {
     internal class Program
     {
-        private static async Task Main()
+        public static async Task Main()
         {
             Console.Title = "KitchenConsole";
 
             await new HostBuilder()
+                .ConfigureAppConfiguration((hostContext, config) =>
+                {
+                    config.SetBasePath(Environment.CurrentDirectory);
+                    config.AddJsonFile("appsettings.json", optional: false);
+                    config.AddJsonFile($"appsettings.{hostContext.HostingEnvironment.EnvironmentName}.json", optional: true);
+                    config.AddEnvironmentVariables();
+                })
                .ConfigureLogging((hostContext, logging) =>
                {
                    logging.AddConsole();
                })
-              .ConfigureServices((hostContext, services) =>
-              {
-                  services.AddJustSaying(config =>
-                  {
-                      config.Client(x =>
-                      {
-                          // Use the simulated pafortin/goaws docker image
-                          x.WithServiceUrl("http://localhost:4100");
+               .ConfigureServices((hostContext, services) =>
+               {
+                   var awsRegion = hostContext.Configuration.GetValue<string>("AWSRegion");
+                   var awsServiceUrl = hostContext.Configuration.GetValue<string>("AWSServiceUrl");
+                   services.AddJustSaying(config =>
+                   {
+                       config.Client(x =>
+                       {
+                           if (!string.IsNullOrWhiteSpace(awsServiceUrl))
+                           {
+                               // The AWS client sdk allows specifying a custom http endpoint.
+                               // For testing purposes it is useful to specify a value that
+                               // points to a docker image such as `p4tin/goaws` or `localstack/localstack`
+                               x.WithServiceUrl(awsServiceUrl);
+                           }
+                           else
+                           {
+                               // The real AWS environment will require some means of authentication
+                               //x.WithBasicCredentials("###", "###");
```

Comment:
If you add _UserSecrets_ to the config providers, you could load these from config without needing to have it commented-out with placeholders.

Reply:
Yeah im a bit torn - im trying to be careful to leave the sample app in the simplest possible state where people can easily see what they might need to do or change in order to get it running on their box. Some have keys, others env variables etc. Not sure whats best.
Can go with the secrets if you think that's best but I think its a bit of a distraction from demonstrating the bus.

Reply:
on a similar note i was thinking of leaving a line like this for those who configure the region via environment variables too..?
```cs
//x.WithRegion(new Amazon.Runtime.EnvironmentVariableAWSRegion().Region);
```

Reply:
I guess that's more of an advanced use case? We wouldn't want to balloon the samples too much with all the possible things you _could_ do. Also if you were using environment variables, you'd probably be using the .NET provider for that and binding them to `IConfiguration` somehow I'd have thought.

Reply:
Yeah and it feels a bit orthogonal to the actual demonstration of your library. Its more demonstrating aspnet core configuration and the aws sdk